core/services/workflows/syncer/workflow_registry.go

```diff
@@ -353,13 +407,192 @@ func (w *workflowRegistry) readRegistryEvents(ctx context.Context, don capabilit
 			}
 
 			for _, event := range events {
-				err := w.handler.Handle(ctx, event.Event)
+				w.eventCh <- event.Event
+			}
+		}
+	}
+}
+
+// syncUsingEventStrategy syncs workflow registry contract state by watching for events on the contract.
+// It first loads the workflow metadata from the contract state as WorkflowRegistered events,
+// and then starts a goroutine with one loop for handling the events.
+func (w *workflowRegistry) syncUsingEventStrategy(ctx context.Context, don capabilities.DON, reader ContractReader) {
+	w.lggr.Debugw("Loading initial workflows for DON", "DON", don.ID)
+
+	workflowMetadata, loadWorkflowsHead, err := w.getWorkflowMetadata(ctx, don, reader)
+	// TODO: if this fails, we should still start the event looping
+	if err != nil {
+		w.lggr.Errorw("failed to load initial workflows", "err", err)
+		return
+	}
+
+	w.lggr.Debugw("Rehydrating existing workflows", "len", len(workflowMetadata))
+	for _, workflow := range workflowMetadata {
+		toRegisteredEvent := WorkflowRegistryWorkflowRegisteredV1{
+			WorkflowID:    workflow.WorkflowID,
+			WorkflowOwner: workflow.Owner,
+			DonID:         workflow.DonID,
+			Status:        workflow.Status,
+			WorkflowName:  workflow.WorkflowName,
+			BinaryURL:     workflow.BinaryURL,
+			ConfigURL:     workflow.ConfigURL,
+			SecretsURL:    workflow.SecretsURL,
+		}
+		w.eventCh <- workflowAsEvent{
+			Data:      toRegisteredEvent,
+			EventType: WorkflowRegisteredEvent,
+		}
+		if err != nil {
+			w.lggr.Errorf("failed to handle workflow registration: %s", err)
+		}
+	}
+
+	// Poll for all workflow related events
+	ets := []WorkflowRegistryEventType{
+		ForceUpdateSecretsEvent,
+		WorkflowActivatedEvent,
+		WorkflowDeletedEvent,
+		WorkflowPausedEvent,
+		WorkflowRegisteredEvent,
+		WorkflowUpdatedEvent,
+	}
+
+	w.wg.Add(1)
+	defer w.wg.Done()
+	w.readRegistryEventsLoop(ctx, ets, don, reader, loadWorkflowsHead.Height)
+}
+
+// workflowMetadataToEvents compares the workflow registry workflow metadata state against the engine registry's state.
+// Differences are handled by the event handler by creating events that are sent to the events channel for handling.
+func (w *workflowRegistry) workflowMetadataToEvents(ctx context.Context, workflowMetadata []GetWorkflowMetadata, donID uint32) {
+	var events []workflowAsEvent
+
+	// Keep track of which of the engines in the engineRegistry have been touched
+	seenMap := map[string]bool{}
+
+	for _, wfMeta := range workflowMetadata {
+		// TODO: ensure that the WorkflowRegisteredEvent clears the engine registry as the very last step
+		engine, getErr := w.engineRegistry.Get(EngineRegistryKey{Owner: wfMeta.Owner, Name: wfMeta.WorkflowName})
+		engineKey := wfMeta.WorkflowName + hex.EncodeToString(wfMeta.Owner)
+
+		// if there is no engine in the engine registry
```

Comment:
Callout here: The way the handler currently works is that it will do work on a Registered event in a paused status. It has no checkpointing, which is a concept that will happen in the handler refactor. Because it never starts the engine, this registered handling will fire every tick, causing URLs to be fetched every tick.


Paths forward are:
1. ignore this until the fast follow next PR to refactor handler
2. add some complexity to this PR by making a few select handler changes
3. not process paused workflows by having a check here on the status
@cedric-cordenier WDYT?

Reply:
Went forward with option 3.